### PR TITLE
Pico GPT v2 code for #51. Includes changes for #61

### PR DESCRIPTION
**Pull request by hmf**
Changes for #51 and #61.

### CODE REVIEW

**Review by sbrunk — COMMENTED** [review]
I think this got in with the latest commit by accident?

File: tmp.txt
Comment (sbrunk) [inline tmp.txt:1]: I think this got in with the latest commit by accident?

File: examples/src/main/scala/gpt/v2.scala [hunk 1]
```diff
@@ -0,0 +1,1012 @@
...
+    * that usually holds the variable name and class parameter values. We also add
+    * the number of tensor parameter amd their value in the leaf modules. For the 
+    * other modules the sum of the number of tensor parameters are shown. 
+    * 
+    * Use this output to "debug" your networks
+    *
+    * @param m
+    * @return string
+    */
+  def doModuleInfoString(m:Module, indent: Int): String =
```
Comment (sbrunk) [inline examples/src/main/scala/gpt/v2.scala:91]: This looks pretty useful in general. Do you think we could move that into `Module`?

File: tmp.txt
Comment (hmf) [reply tmp.txt:1]: Correct. Will delete.

**Comment by sbrunk** [comment]
I've just run `sbt 'headerCreateAll ; scalafmtAll'` to make the CI linting checks happy. I'd suggest we merge this now to get your op and improvements in and move the utils methods into core in a subsequent PR.

Would that be OK with you?

**Comment by hmf** [comment]
@sbrunk Yes. It is ok with me.

**Pull request merged**